### PR TITLE
381 - convert active hash to active record

### PR DESCRIPTION
**Pull request by jmkoni**
Not 100% done yet, still need to at least do a feature test. Since this is my first deep dive into bridge_troll, wanted to get a review a bit earlier. Plus, this touched quite a bit of the code.

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -362,6 +362,16 @@ def update_rsvp_counts
     )
   end
 
+  def current_state
```
Comment (tjgrathwell) [inline app/models/event.rb]: this replicates methods provided by `enum current_state` and is blowing everything up. remove it?

File: app/assets/javascripts/levels.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (tjgrathwell) [inline app/assets/javascripts/levels.coffee]: plz remove extra scaffolded files

File: app/models/level.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Level < ActiveRecord::Base
+  belongs_to :course
+  validates :num, presence: true, numericality: true, uniqueness: { scope: :course_id }
+  validates :color, presence: true, uniqueness: { scope: :course_id }
+  validates :title, presence: true
+  validates :level_description, presence: true
+
+  # level_description is an array that is getting saved as a string
+  # this parses it back as an array
+  def description
```
Comment (tjgrathwell) [inline app/models/level.rb]: there's probably a more elegant way to do this with [serialize](http://apidock.com/rails/ActiveRecord/Base/serialize/class)

File: spec/features/events/event_rsvp_request_spec.rb [hunk 4]
```diff
@@ -160,15 +163,15 @@
     describe "a new learn rsvp" do
       it "should show rails levels for rails events" do
         visit learn_new_event_rsvp_path(@event)
-        expect(page).to have_content Course.find_by_name('RAILS').levels[0][:title]
+        expect(page).to have_content @course.levels[0][:title]
       end
 
       it "should show frontend levels for frontend events" do
```
Comment (tjgrathwell) [inline spec/features/events/event_rsvp_request_spec.rb]: probably don't need this to be a separate test if it's now just reading the value out of the DB

File: app/models/event.rb (hunk 1 above)
Comment (jmkoni) [inline app/models/event.rb]: the weird thing is... I didn't actually add this. I merged in master prior to doing the PR, but if it was there, then it shouldn't be a diff. I will remove it either way.

**Comment by jmkoni** [comment]
I've made some of the changes, but I'm noticing that there are some weird merge things going on (changes I didn't make that are now causing problems). I'm going to start over from a fresh copy of master and apply my actual changes there. I'm right now getting test failures that I'm *pretty* sure shouldn't be caused by code I added.

**Comment by jmkoni** [comment]
So with that, I'm going to close this PR, but the changes you mentioned in the comments will be on my next. Gotta get a fresh branch :)

**Comment by lilliealbert** [comment]
Cool! In the future, I think you should usually be able to rebase your diff onto master and then force push over your branch to get your PR cleaned up. I don't know if it'll help if you're seeing weird git stuff, but we do have a little guide here: http://railsbridge.github.io/bridge_troll/

(As a project, we generally try to avoid merge commits and rebase pretty aggressively 😄 )

**Comment by jmkoni** [comment]
yeah I thought I had pulled recently and was VERY mistaken :P

**Pull request closed without merging**